### PR TITLE
Modify playbooks to work with satellite server

### PR DESCRIPTION
**Pull request by jasonkha**

### CODE REVIEW

File: playbooks/roles/ambari-agent/tasks/main.yml [hunk 1]
```diff
@@ -28,6 +28,11 @@
         (ansible_python_version is version_compare('2.7', '>=') and ((ansible_os_family|lower == "redhat" and ansible_distribution_major_version|int >= 7) or
                                                                      (ansible_os_family|lower == "suse" and ansible_distribution_major_version|int >= 12) or
                                                                      (ansible_os_family|lower == "redhat" and ansible_distribution|lower == "amazon")))
+- name: Comment out package_version = None in script.py
```
Comment (zer0glitch) [inline playbooks/roles/ambari-agent/tasks/main.yml:31]: What is this change for?

File: playbooks/roles/ambari-config/templates/vdf-HDF-part-repository-info.j2 [hunk 2]
```diff
@@ -1,16 +1,15 @@
     <os family="{{ baseurl_os_family }}">
       <package-version>{{ hdf_version|replace('-','_') }}_*</package-version>
       <repo>
-        <baseurl>{{ hdf_main_repo_url }}</baseurl>
```
Comment (zer0glitch) [inline playbooks/roles/ambari-config/templates/vdf-HDF-part-repository-info.j2:4]: I have another work around for this to make this work in a more generic environment.

File: playbooks/roles/ambari-config/templates/vdf-HDP-part-repository-info.j2 [hunk 3]
```diff
@@ -1,30 +1,30 @@
     <os family="{{ baseurl_os_family }}">
       <package-version>{{ hdp_version|replace('-','_') }}_*</package-version>
       <repo>
-        <baseurl>{{ hdp_main_repo_url }}</baseurl>
```
Comment (zer0glitch) [inline playbooks/roles/ambari-config/templates/vdf-HDP-part-repository-info.j2:4]: I have a generic work around for this

File: playbooks/roles/ambari-config/vars/redhat.yml [hunk 4]
```diff
@@ -2,12 +2,12 @@
 repo_os_family: "centos{{ ansible_distribution_major_version }}"
 baseurl_os_family: "redhat{{ ansible_distribution_major_version }}"
 
-hdp_main_repo_url: "{{ repo_base_url }}/HDP/{{ repo_os_family }}/{{ hdp_major_version }}.x/updates/{{ hdp_version }}"
```
Comment (zer0glitch) [inline playbooks/roles/ambari-config/vars/redhat.yml:5]: I have a generic workaround for this

File: playbooks/roles/ambari-server/tasks/main.yml [hunk 5]
```diff
@@ -22,6 +22,20 @@
     state: latest
   notify: Reload systemd
 
+- name: Create missing ambari-server symlinks
```
Comment (zer0glitch) [inline playbooks/roles/ambari-server/tasks/main.yml:25]: This may be a version issue, I am not seeing this with the current install

**Review by zer0glitch — CHANGES_REQUESTED** [review]

File: playbooks/roles/ambari-agent/tasks/main.yml (hunk 1 above)
Comment (jasonkha) [reply playbooks/roles/ambari-agent/tasks/main.yml:31]: I had issues with certain components failing to install. Error message was:
raise Fail("Cannot match package for regexp name {0}. Available packages: {1}".format(name, self.available_packages_in_repos)) 

Hortonworks Support pointed me to the thread below, which fixed the issue.
https://community.hortonworks.com/questions/203492/new-install-hdp-2650-292-via-ambari-2620-155-zooke.html

File: playbooks/roles/ambari-server/tasks/main.yml (hunk 5 above)
Comment (jasonkha) [reply playbooks/roles/ambari-server/tasks/main.yml:25]: This fixed the ImportError: No module named ambari_commons.exceptions.

Comment (jasonkha) [reply playbooks/roles/ambari-server/tasks/main.yml]: This was the fix for the ImportError: No module named ambari_commons.exceptions that the support agent found.

**Pull request closed without merging**